src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java

```diff
@@ -67,6 +71,49 @@
 		return map;
 	}
 
+	protected PodSpecBuilder createPodSpecBuilder(KubernetesDeployerProperties properties) {
+		PodSpecBuilder podSpecBuilder = new PodSpecBuilder();
+
+		// Add image secrets if set
+		if (properties.getImagePullSecret() != null) {
+			podSpecBuilder.addNewImagePullSecret(properties.getImagePullSecret());
+		}
+
+		return podSpecBuilder;
+	}
+
+	protected Container addMemoryCpu(AppDeploymentRequest request, Container container,
```

Comment:
Can this method be more generic saying `addResourceLimits` instead of `addMemoryCpu`?